libs/dh/admin/feature-user-management-new/src/lib/edit/edit.component.ts

```diff
@@ -127,32 +127,56 @@ export class DhEditUserComponent {
     });
   }
 
-  save() {
+  async save() {
+    const id = this.id();
     if (
       this.userInfoForm.invalid ||
-      this.updateUserRoles?.actors.some((actor) => !actor.atLeastOneRoleIsAssigned)
+      this.updateUserRoles?.actors.some((actor) => !actor.atLeastOneRoleIsAssigned) ||
+      id === undefined
     ) {
       return;
     }
 
-    if (this.userInfoForm.pristine) {
-      return this.close();
-    }
+    const { firstName, lastName, phoneNumber } = this.userInfoForm.getRawValue();
 
-    const user = this.user();
+    const phoneParts = phoneNumber.split(' ');
+    const [prefix, ...rest] = phoneParts;
+    const formattedPhoneNumber = `${prefix} ${rest.join('')}`;
 
-    const { firstName, lastName, phoneNumber } = this.userInfoForm.getRawValue();
+    const updateActorUserRolesInput: UpdateActorUserRolesInput[] = this.updateUserRoles
+      ? this.updateUserRoles?.actors.map((actor) => {
+          return {
+            actorId: actor.id,
+            assignments: actor.userRolesToUpdate,
+          };
+        })
+      : [];
 
-    if (
-      firstName === user?.firstName &&
-      lastName === user?.lastName &&
-      phoneNumber === user?.phoneNumber &&
-      this.updateUserRoles === null
-    ) {
-      return this.close();
+    const result = await this.editUserMutation.mutate({
+      refetchQueries: [GetUsersDocument, GetUserByIdDocument],
+      variables: {
+        updateRolesInput: {
+          userId: id,
+          input: updateActorUserRolesInput,
+        },
+        updateUserInput: {
+          userId: id,
+          userIdentityUpdateDto: {
+            firstName,
+            lastName,
+            phoneNumber: formattedPhoneNumber,
+          },
+        },
+      },
+    });
+
+    if (result.data?.updateUserIdentity.success && result.data?.updateUserRoleAssignment.success) {
+      this.showSuccessToast();
     }
```

Comment:
suggestion: in some cases `onCompleted` and `onError` callbacks on the mutation are used instead. For consistency sake, should we stick to only one way?

Reply:
I'm fine with both solutions.